forc-plugins/forc-debug/src/server/handlers/handle_launch.rs

```diff
@@ -0,0 +1,216 @@
+                    &pkg_to_debug.bytecode.bytes,
+                    offset,
+                    test_setup.clone(),
+                    test_entry,
+                    name.clone(),
+                ))
+            })
+            .collect();
+        self.state.init_executors(executors);
+
+        // Start debugging
+        self.state.update_vm_breakpoints();
+        while let Some(executor) = self.state.executors.first_mut() {
+            executor.interpreter.set_single_stepping(false);
+
+            match executor.start_debugging()? {
+                DebugResult::TestComplete(result) => {
+                    self.state.test_results.push(result);
+                }
+                DebugResult::Breakpoint(pc) => {
+                    return self.stop_on_breakpoint(pc);
+                }
+            };
+            self.state.executors.remove(0);
+        }
+
+        self.log_test_results();
+        Ok(false)
+    }
+
+    /// Builds the tests at the given [PathBuf] and stores the source maps.
+    pub(crate) fn build_tests(&mut self) -> Result<(BuiltPackage, TestSetup), AdapterError> {
+        if let Some(pkg) = &self.state.built_package {
+            if let Some(setup) = &self.state.test_setup {
+                return Ok((pkg.clone(), setup.clone()));
+            }
+        }
+
+        // 1. Build the packages
+        let manifest_file = forc_pkg::manifest::ManifestFile::from_dir(&self.state.program_path)
+            .map_err(|err| AdapterError::BuildFailed {
+                phase: "read manifest file".into(),
+                reason: err.to_string(),
+            })?;
+        let pkg_manifest: PackageManifestFile =
+            manifest_file
+                .clone()
+                .try_into()
+                .map_err(|err: anyhow::Error| AdapterError::BuildFailed {
+                    phase: "package manifest".into(),
+                    reason: err.to_string(),
+                })?;
+        let member_manifests =
+            manifest_file
+                .member_manifests()
+                .map_err(|err| AdapterError::BuildFailed {
+                    phase: "member manifests".into(),
+                    reason: err.to_string(),
+                })?;
+        let lock_path = manifest_file
+            .lock_path()
+            .map_err(|err| AdapterError::BuildFailed {
+                phase: "lock path".into(),
+                reason: err.to_string(),
+            })?;
+        let build_plan = forc_pkg::BuildPlan::from_lock_and_manifests(
+            &lock_path,
+            &member_manifests,
+            false,
+            false,
+            Default::default(),
+        )
+        .map_err(|err| AdapterError::BuildFailed {
+            phase: "build plan".into(),
+            reason: err.to_string(),
+        })?;
+
+        let project_name = pkg_manifest.project_name();
+
+        let outputs =
+            std::iter::once(build_plan.find_member_index(project_name).unwrap()).collect();
```

Comment:
is it possible to return an error instead of unwraping here? Or at least have it be an expect?